### DISCUSSION TITLE
Remove some blocks and then use standard PyMuPDF get_text functionality?

### DISCUSSION THREAD

**Discussion by stevesimmons** [pymupdf/PyMuPDF#2079] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I want to extract text from PDFs where some pages have text overlays that need to be ignored.

**Describe the solution you'd like**
I would like to scan the blocks on a page, delete (or otherwise mark as "ignore") the ones I don't want, and then use `page.get_text(clip=rect)` on the target regions for the remaining blocks.

**Describe alternatives you've considered**
Alternatively stream the elements of a page into a new empty page, filtering out the blocks I don't need. And then do the text extraction on the filtered page.

**Additional context**
Any suggestions would be most welcome. I would like to solve 100% of this problem with PyMuPDF because it is such a great tool.

**Comment by JorjMcKie** [comment, marked as answer]
The most challenging part is how you want to identify undesired text. If you can do that, no additional technique is required: simply ignore it on encounter while extracting text.
But maybe I miss some aspect of your requirement.

When you do `page.get_text("dict", ...)` you will be given all text with all available meta-information, from font name, ~ properties, ~ size, text color, position information, writing direction.
What else do you need to for identification?

**Comment by JorjMcKie** [comment]
Carefully looking again, I am not sure what you technically mean by text **_"overlays"_**. Just text overlapping other text?

**Comment by stevesimmons** [comment]
Ah, good suggestion! I am too used to `page.get_text(clip=...)` magically bringing my text back in a single string.

I'll look at the `dict` option now and use the other properties to filter out the text I don't want. The bits I don't want are a different font size, so irrespective of the clip window I will still be able to filter them out. 

Thank you
